File: src/Module/Ticket/TicketEntity.java
package Module.Ticket;

import AppStart.RESTTimestamp;
import Module.Employee.EmployeeEntity;
import Module.Employee.EmployeeModel;
import Module.Location.LocationEntity;
import Module.Location.LocationModel;
import Module.TicketImage.TicketImageEntity;
import Module.TicketImage.TicketimageModel;
import Module.TicketRead.TicketReadEntity;
import Module.TicketRead.TicketreadModel;
import Module.TicketRelater.TicketRelaterEntity;
import Module.TicketRelater.TicketrelaterModel;
import Module.TicketThread.TicketThreadEntity;
import Module.TicketThread.TicketthreadModel;
import Module.TicketAttribute.TicketAttributeEntity;
import Module.TicketAttribute.TicketattributeModel;

import java.io.Serializable;
import java.io.UnsupportedEncodingException;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class TicketEntity implements Serializable {
    public int id;
    public String content;
    public String subject;
    public Integer createEmployeeId;
    public Integer status;
    public Integer priority;
    public String deadline;
    public Integer assignedEmployeeId;
    public Byte rating;
    public Integer locationId;
    public String resolvedTime;
    public String closedTime;
    public String createdTime;
    public String updatedTime;
    public String deletedTime;
    public EmployeeEntity createEmployeeEntity;
    public EmployeeEntity assignedEmployeeEntity;
    public LocationEntity locationEntity;
    public List<TicketAttributeEntity> ticketAttributeEntities;
    public List<TicketImageEntity> ticketImageEntities;
    public List<TicketReadEntity> ticketReadEntities;
    public List<TicketRelaterEntity> ticketRelaterEntities;
    public List<TicketThreadEntity> ticketThreadEntities;

    public TicketEntity() {
    }

    public TicketEntity(TicketModel ticketModel, Object... objects) {
        this.id = ticketModel.getId();
        this.content = ticketModel.getContent();
        this.subject = ticketModel.getSubject();
        this.createEmployeeId = ticketModel.getCreateEmployeeId();
        this.status = ticketModel.getStatus();
        this.priority = ticketModel.getPriority();
        this.deadline = ticketModel.getDeadline().toString();
        this.assignedEmployeeId = ticketModel.getAssignedEmployeeId();
        this.rating = ticketModel.getRating();
        this.locationId = ticketModel.getLocationId();
        this.resolvedTime =  RESTTimestamp.toString(ticketModel.getResolvedTime());
        this.closedTime =RESTTimestamp.toString(ticketModel.getClosedTime());
        this.createdTime =  RESTTimestamp.toString(ticketModel.getCreatedTime());
        this.updatedTime =  RESTTimestamp.toString(ticketModel.getUpdatedTime());
        this.deletedTime = RESTTimestamp.toString(ticketModel.getDeletedTime());
        for (Object object : objects) {
            if (object instanceof EmployeeModel) {
                this.createEmployeeEntity = new EmployeeEntity((EmployeeModel) object);
            } else if (object instanceof LocationModel) {
                this.locationEntity = new LocationEntity((LocationModel) object);
            } else if (object instanceof Collection) {
                for (Object o : (Collection<Object>) object) {
                    if (o instanceof TicketattributeModel) {
                        this.ticketAttributeEntities = ((Collection<TicketattributeModel>) object).parallelStream().map(TicketAttributeEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TicketimageModel) {
                        this.ticketImageEntities = ((Collection<TicketimageModel>) object).parallelStream().map(TicketImageEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TicketreadModel) {
                        this.ticketReadEntities = ((Collection<TicketreadModel>) object).parallelStream().map(TicketReadEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TicketrelaterModel) {
                        this.ticketRelaterEntities = ((Collection<TicketrelaterModel>) object).parallelStream().map(TicketRelaterEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TicketthreadModel) {
                        this.ticketThreadEntities = ((Collection<TicketthreadModel>) object).parallelStream().map(TicketThreadEntity::new).collect(Collectors.toList());
                        break;
                    }
                }
            }
        }
    }

    public TicketModel toModel() {
        TicketModel ticketModel = new TicketModel();
        ticketModel.setId(id);
        try {
            ticketModel.setSubject(new String(subject.getBytes(), "UTF-8"));
            ticketModel.setContent(new String(content.getBytes(), "UTF-8"));
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
        }
        ticketModel.setCreateEmployeeId(createEmployeeId);
        ticketModel.setStatus(status);
        ticketModel.setPriority(priority);
        ticketModel.setDeadline(RESTTimestamp.Parse(deadline));
        ticketModel.setAssignedEmployeeId(assignedEmployeeId);
        ticketModel.setRating(rating);
        ticketModel.setLocationId(locationId);
        ticketModel.setResolvedTime(RESTTimestamp.Parse(resolvedTime));
        ticketModel.setClosedTime(RESTTimestamp.Parse(closedTime));
        ticketModel.setCreatedTime(RESTTimestamp.Parse(createdTime));
        ticketModel.setUpdatedTime(RESTTimestamp.Parse(updatedTime));
        ticketModel.setDeletedTime(RESTTimestamp.Parse(deletedTime));
        return ticketModel;
    }

    public TicketModel toModel(List<TicketImageEntity> ticketImageEntities) {
        TicketModel ticketModel = toModel();
        if(ticketImageEntities != null){
            Collection<TicketimageModel> ticketimageModels = ticketImageEntities.parallelStream().map(TicketImageEntity::toModel).collect(Collectors.toList());
            ticketModel.setTicketimagesById(ticketimageModels);
        }
        return ticketModel;
    }
}

File: web/Angular4/src/app/Modules/Ticket/Ticket.Entity.ts
import {EmployeeEntity} from "../Employee/Employee.Entity";
import {LocationEntity} from "../Location/Location.Entity";
import {TicketAttributeEntity} from "../TicketAttribute/TicketAttribute.Entity";
import {TicketImageEntity} from "../TicketImage/TicketImage.Entity";
import {TicketReadEntity} from "../TicketRead/TicketRead.Entity";
import {TicketRelaterEntity} from "../TicketRelater/TicketRelater.Entity";
import {TicketThreadEntity} from "../TicketThread/TicketThread.Entity";

export class TicketEntity {
    id: number;
    content: string;
    subject: string;
    createEmployeeId: number;
    status: number;
    priority: number;
    deadline: string;
    assignedEmployeeId: number;
    rating: number;
    locationId: number;
    resolvedTime: string;
    closedTime: string;
    createdTime: string;
    updatedTime: string;
    deletedTime: string;
    createEmployeeEntity: EmployeeEntity;
    assignedEmployeeEntity: EmployeeEntity;
    locationEntity: LocationEntity;
    ticketAttributeEntities: TicketAttributeEntity[];
    ticketImageEntities: TicketImageEntity[];
    ticketReadEntities: TicketReadEntity[];
    ticketRelaterEntities: TicketRelaterEntity[];
    ticketThreadEntities: TicketThreadEntity[];
    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;

    constructor(data: any = null) {
        if (data == null) {
            this.id = null;
            this.content = null;
            this.subject = null;
            this.createEmployeeId = null;
            this.status = null;
            this.priority = null;
            this.deadline = null;
            this.assignedEmployeeId = null;
            this.rating = null;
            this.locationId = null;
            this.resolvedTime = null;
            this.closedTime = null;
            this.createdTime = null;
            this.updatedTime = null;
            this.deletedTime = null;
            this.createEmployeeEntity = new EmployeeEntity();
            this.assignedEmployeeEntity = new EmployeeEntity();
            this.ticketAttributeEntities = [];
            this.ticketImageEntities = [];
            this.ticketReadEntities = [];
            this.ticketRelaterEntities = [];
            this.ticketThreadEntities = [];
        } else {
            this.id = data.id;
            this.content = data.content;
            this.subject = data.subject;
            this.createEmployeeId = data.createEmployeeId;
            this.status = data.status;
            this.priority = data.priority;
            this.deadline = data.deadline;
            this.assignedEmployeeId = data.assignedEmployeeId;
            this.rating = data.rating;
            this.locationId = data.locationId;
            this.resolvedTime = data.resolvedTime;
            this.closedTime = data.closedTime;
            this.createdTime = data.createdTime;
            this.updatedTime = data.updatedTime;
            this.deletedTime = data.deletedTime;
            this.createEmployeeEntity = data.createEmployeeEntity == null ? new EmployeeEntity() : data.createEmployeeEntity;
            this.assignedEmployeeEntity = data.assignedEmployeeEntity == null ? new EmployeeEntity() : data.assignedEmployeeEntity;
            this.locationEntity = data.locationEntity;
            ;
            if (data.ticketAttributeEntities != null) {
                this.ticketAttributeEntities = [];
                for (let item of data.ticketAttributeEntities) {
                    this.ticketAttributeEntities.push(new TicketAttributeEntity(item));
                }
            } else data.ticketAttributeEntities = [];
            if (data.ticketImageEntities != null) {
                this.ticketImageEntities = [];
                for (let item of data.ticketImageEntities) {
                    this.ticketImageEntities.push(new TicketImageEntity(item));
                }
            } else data.ticketImageEntities = [];
            if (data.ticketReadEntities != null) {
                this.ticketReadEntities = [];
                for (let item of data.ticketReadEntities) {
                    this.ticketReadEntities.push(new TicketReadEntity(item));
                }
            } else data.ticketReadEntities = [];
            if (data.ticketRelaterEntities != null) {
                this.ticketRelaterEntities = [];
                for (let item of data.ticketRelaterEntities) {
                    this.ticketRelaterEntities.push(new TicketRelaterEntity(item));
                }
            } else data.ticketRelaterEntities = [];
            if (data.ticketThreadEntities != null) {
                this.ticketThreadEntities = [];
                for (let item of data.ticketThreadEntities) {
                    this.ticketThreadEntities.push(new TicketThreadEntity(item));
                }
            } else data.ticketThreadEntities = [];
        }
        this.IsEdit = false;
    }
}